mobile/src/employee-details/employee-details-events-list.tsx

```diff
@@ -122,15 +122,39 @@ export class EmployeeDetailsEventsList extends Component<EmployeeDetailsEventsLi
     private descriptionStatus(event: CalendarEvent): string {
         let description = '';
 
-        if (event.isRequested) {
-            description = `requests ${event.type.toLowerCase()}`;
-        } else if (event.isApproved || event.isProcessed) {
-            const prefix = event.dates.endDate.isAfter(moment(), 'date') ? 'has coming ' : 'on ';
-            description = prefix + event.type.toLowerCase();
-        } else if (event.isCompleted) {
-            description = `has completed ${event.type.toLowerCase()}`;
+        switch (this.preprocessStatus(event)) {
+            case CalendarEventStatus.Requested:
+                description = `requests ${event.type.toLowerCase()}`;
+                break;
+            case CalendarEventStatus.Approved:
+                const prefix = event.dates.endDate.isAfter(moment(), 'date') ? 'has coming ' : 'on ';
+                description = prefix + event.type.toLowerCase();
+                break;
+            case CalendarEventStatus.Completed:
+                description = `has completed ${event.type.toLowerCase()}`;
+                break;
+            default:
+                break;
         }
 
         return description;
     }
+
+    //----------------------------------------------------------------------------
+    // noinspection JSMethodCanBeStatic
+    private preprocessStatus(event: CalendarEvent): CalendarEventStatus {
+        if (!event.isVacation) {
+            return event.status;
+        }
+
+        switch (event.status) {
+            case CalendarEventStatus.Requested:
```

Comment:
You perceived Alex's words too literally, as for me, but ok :)

Reply:
Well, this leaves the descriptionStatus function intact, I didn't want to deal with if's conditions there.